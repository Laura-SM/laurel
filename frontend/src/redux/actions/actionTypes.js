export default {
  LOAD_PLANTS: 'LOAD_PLANTS',
  ADD_PLANT: 'ADD_PLANT',
  UPDATE_PLANT: 'UPDATE_PLANT',
  DELETE_PLANT: 'DELETE_PLANT',
  LOAD_PLANT: 'LOAD_PLANT',
  PLANTS_ERROR: 'PLANTS_ERROR',

  GET_USER: 'GET_USER',
  USER_ERROR: 'USER_ERROR',
};
