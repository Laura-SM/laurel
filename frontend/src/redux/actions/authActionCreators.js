import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function signInUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signin`, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}

export function signUpUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/signup`, user);
      dispatch({
        type: actionTypes.SIGNUP_USER,
        user: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
