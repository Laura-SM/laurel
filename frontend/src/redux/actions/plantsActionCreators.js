import axios from 'axios';
import actionTypes from './actionTypes';
import {env} from '../../../.env.js';

const url = env.REACT_APP_PORT;

export function loadPlants() {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants`);
      dispatch({
        type: actionTypes.LOAD_PLANTS,
        plants: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function addPlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.post(`${url}/plants`, plant);
      dispatch({
        type: actionTypes.ADD_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function deletePlant(plantId) {
  return async dispatch => {
    try {
      await axios.delete(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.DELETE_PLANT,
        plantId,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function updatePlant(plant) {
  return async dispatch => {
    try {
      const {data} = await axios.put(`${url}/plants/${plant._id}`, plant);
      dispatch({
        type: actionTypes.UPDATE_PLANT,
        plant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}

export function loadPlant(plantId) {
  return async dispatch => {
    try {
      const {data} = await axios(`${url}/plants/${plantId}`);
      dispatch({
        type: actionTypes.LOAD_PLANT,
        selectedPlant: data,
      });
    } catch (error) {
      dispatch({
        type: 'PLANTS_ERROR',
      });
    }
  };
}
