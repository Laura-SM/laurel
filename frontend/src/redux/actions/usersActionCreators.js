import axios from 'axios';
import actionTypes from './actionTypes';

const url = 'http://192.168.8.104:2021/login';

export function loadUser(user) {
  return async dispatch => {
    try {
      const {data} = await axios.post(url, user);
      dispatch({
        type: actionTypes.SIGNIN_USER,
        userTokens: data,
      });
    } catch (error) {
      dispatch({
        type: actionTypes.USER_ERROR,
      });
    }
  };
}
