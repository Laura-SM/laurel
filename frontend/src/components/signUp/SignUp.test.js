import 'react-native';
import React from 'react';
import SignUp from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import * as actions from '../../redux/actions/authActionCreators';
import {Provider} from 'react-redux';
import {render, fireEvent} from '@testing-library/react-native';

describe('Given a SignUp component', () => {
  const mockStore = Store([thunk]);
  let navigation;
  let myStore;

  beforeEach(() => {
    jest.spyOn(actions, 'signUpUser').mockReturnValueOnce({type: ''});
    myStore = mockStore({});
    navigation = {
      navigate: jest.fn(),
    };
  });

  describe('When invoques SignIn function', () => {
    test('Then SignIn component renders correctly', () => {
      const wrapper = render(
        <Provider store={myStore}>
          <SignUp />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When a new user is sign up', () => {
    test('Then navigation is invoked ', () => {
      myStore = mockStore({
        user: {email: 'laura@email.com', password: '1a'},
      });
      render(
        <Provider store={myStore}>
          <SignUp navigation={navigation} />
        </Provider>,
      );

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });

  describe('When Sign Up button is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <SignUp navigation={navigation} />
        </Provider>,
      );
      const signIn = getByTestId('signUp');
      fireEvent.press(signIn);

      expect(actions.signUpUser).toHaveBeenCalled();
    });
  });

  describe('When emailInput is typing with laura@email.com', () => {
    test('Then emailInput value is laura@email.com', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <SignUp />
        </Provider>,
      );
      const emailInput = getByTestId('emailInput');
      const email = 'laura@email.com';
      fireEvent.changeText(emailInput, email);

      expect(emailInput.props.value).toBe(email);
    });
  });

  describe('When passwordInput is typing with 123', () => {
    test('Then passwordInput value is 123', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <SignUp />
        </Provider>,
      );
      const passwordInput = getByTestId('passwordInput');
      const password = '123';
      fireEvent.changeText(passwordInput, password);

      expect(passwordInput.props.value).toBe(password);
    });
  });

  describe('When Sign In pressable text is pressed', () => {
    test('Then navigation.navigate is invoked', () => {
      const {getByTestId} = render(
        <Provider store={myStore}>
          <SignUp navigation={navigation} />
        </Provider>,
      );
      const register = getByTestId('goSignIn');
      fireEvent.press(register);

      expect(navigation.navigate).toHaveBeenCalled();
    });
  });
});
