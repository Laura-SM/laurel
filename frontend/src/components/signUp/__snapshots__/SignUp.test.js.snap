// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a SignUp component When invoques SignIn function Then SignIn component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#D6EADF",
      "flex": 1,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://i.ibb.co/pbFgzLP/laurel-logo.png",
      }
    }
    style={
      Object {
        "height": 290,
        "marginTop": 60,
        "width": 320,
      }
    }
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="email"
    rejectResponderTermination={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "margin": 5,
        "paddingHorizontal": 10,
        "width": 350,
      }
    }
    testID="emailInput"
    textContentType="emailAddress"
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "fontSize": 20,
        "height": 50,
        "justifyContent": "center",
        "margin": 5,
        "paddingHorizontal": 10,
        "width": 350,
      }
    }
    testID="passwordInput"
    textContentType="password"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#B8E0D2",
        "borderRadius": 10,
        "height": 65,
        "justifyContent": "center",
        "margin": 20,
        "opacity": 1,
        "width": 150,
      }
    }
    testID="signUp"
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Sign Up
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "margin": 20,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="goSignIn"
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginHorizontal": 20,
          }
        }
      >
        Already a member?
         
        <Text
          style={
            Object {
              "fontSize": 18,
              "textDecorationLine": "underline",
            }
          }
        >
          Sign In
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
